### PR TITLE
Checkpoint update till blockchain state

### PR DESCRIPTION
**Pull request by amityadav0**

### CODE REVIEW

File: packages/algob/src/internal/deployer.ts [hunk 1]
```diff
@@ -104,22 +106,55 @@ export class AlgobDeployerImpl implements AlgobDeployer {
...
-      name, this.loadedAsaDefs[name], flags, this.accountsByName, this.txWriter);
+    let asaInfo = {} as any;
+    try {
+      asaInfo = await this.algoOp.deployASA(
+        name, this.loadedAsaDefs[name], flags, this.accountsByName, this.txWriter);
+    } catch (error) {
+      persistCheckpoint(this.txWriter.scriptName, this.cpData.strippedCP);
+
+      console.log(error);
+      throw new Error();
```
Comment (robert-zaremba) [inline packages/algob/src/internal/deployer.ts:124]: don't throw empty error. You can re-throw existing:

```
throw error
```

File: packages/algob/src/internal/deployer.ts [hunk 2]
```diff
@@ -104,22 +106,55 @@ export class AlgobDeployerImpl implements AlgobDeployer {
...
     this.assertNoAsset(name);
-    const ascInfo = await this.algoOp.deployASC(name, scParams, flags, payFlags, this.txWriter);
+    let ascInfo = {} as any;
+    try {
+      ascInfo = await this.algoOp.deployASC(name, scParams, flags, payFlags, this.txWriter);
+    } catch (error) {
+      persistCheckpoint(this.txWriter.scriptName, this.cpData.strippedCP);
+
+      console.log(error);
+      throw new Error();
```
Comment (robert-zaremba) [inline packages/algob/src/internal/deployer.ts:156]: as above

File: packages/algob/src/internal/deployer.ts [hunk 3]
```diff
@@ -104,22 +106,55 @@ export class AlgobDeployerImpl implements AlgobDeployer {
...
+        name,
+        this.loadedAsaDefs[name],
+        flags,
+        this.accountsByName,
+        asaInfo.assetIndex);
+    } catch (error) {
+      persistCheckpoint(this.txWriter.scriptName, this.cpData.strippedCP);
+
+      console.log(error);
+      throw new Error();
```
Comment (robert-zaremba) [inline packages/algob/src/internal/deployer.ts:140]: ditto

**Review by robert-zaremba — CHANGES_REQUESTED** [review]

File: packages/algob/src/internal/deployer.ts (hunk 1 above)
Comment (amityadav0) [reply packages/algob/src/internal/deployer.ts]: ok

**Review by robert-zaremba — APPROVED** [review]

**Pull request merged**